src/ORM/DataList.php

```diff
@@ -1278,10 +1290,6 @@ private function fetchEagerLoadManyMany(
         $fetchedIDs = [];
         $eagerLoadedLists = [];
 
-        // Get the join records so we can correctly identify which children belong to which parents
-        // This also holds extra fields data
-        $joinRows = DB::query('SELECT * FROM "' . $joinTable . '" WHERE "' . $parentIDField . '" IN (' . implode(',', $parentIDs) . ')');
```

Comment:
Moved to be after fetching the records we care about, and is now filtered and sorted based on that.